### DISCUSSION TITLE
Static texture

### DISCUSSION THREAD

**Discussion by espkk** [DiligentGraphics/DiligentCore#459] [General]
[This](http://diligentgraphics.com/diligent-engine/using-the-api/#Pipeline_Resource_Layout) has an example of texture (`g_StaticTexture`) used with `SHADER_RESOURCE_VARIABLE_TYPE_STATIC`.

When I'm trying that (e.g. by changing type in `Tutorial03_Texturing` and binding it using `IPipelineState::GetStaticVariableByName`) it seems that the texture and its sampler are still associated with SRB, leaving `Diligent Engine: ERROR: No resource is bound to variable 'g_Texture' in shader 'Cube PS' of PSO 'Cube PSO'` in the output. If no SRB is set it results in an assertion instead.

Is it possible to bind texture using PSO or the documentation is just inaccurate there?

**Comment by TheMostDiligent** [comment]
> If no SRB is set it results in an assertion instead.

That is correct. You still need to bind an SRB if there are any resources. The PSO keeps references to static resources, but the SRB still needs to be created and set.

**Comment by espkk** [comment]
Understood. I also figured out that `Diligent Engine: ERROR: No resource is bound to variable 'g_Texture' in shader 'Cube PS' of PSO 'Cube PSO'` comes from that my texture view is `nullptr` when set which most likely technically cannot be detected by `IShaderResourceVariable::Set` 

Thanks for the help!
